middlewares/redirects/index.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-env node */
+
+const { getRedirect } = require('../../src/utils/redirects')
+const { parse } = require('url')
+const { stringify } = require('querystring')
+
+const dev = process.env.NODE_ENV !== 'production'
+
+module.exports = (req, res, next) => {
+  const parsedUrl = parse(req.url, true)
+  const { pathname, query } = parsedUrl
+  const host = req.headers.host
+
+  let [redirectCode, redirectLocation] = getRedirect(host, pathname, {
+    req,
+    dev
+  })
+
+  if (redirectLocation) {
+    // HTTP redirects
+
+    const queryStr = stringify(query)
+    if (queryStr) {
+      redirectLocation += '?' + queryStr
+    }
+    res.writeHead(redirectCode, {
+      Location: redirectLocation
+    })
```

Comment:
We should add `res.end()` here, but we still need to run `next()` after that block for the cases that are not redirected and should be handled by static middleware.

Reply:
You still need to return early here though :snowman: . Otherwise, the next middleware in line will run, which we don't want since `res.end()` has already been called.

Reply:
Yes, you are right, I forgot about it.

Reply:
Done